components/eam/src/physics/cam/shoc_intr.F90

```diff
@@ -1069,11 +1079,28 @@ subroutine shoc_tend_e3sm( &
     enddo
    
     !  Probably need to add deepcu cloud fraction to the cloud fraction array, else would just 
-    !  be outputting the shallow convective cloud fraction 
-
+    !  be outputting the shallow convective cloud fraction
+ 
+    !  Add liq, ice, and precipitating ice fractions here. These are purely
+    !  diagnostic outputs and do not impact the rest of the code. The qi threshold for
+    !  setting ice_cloud_fraction to 1 currently set 1e-8 kg/kg but can be
+    !  changed based on what we think is sensible. 
+
+    liq_cloud_frac = 0.0_r8
+    ice_cloud_frac = 0.0_r8
+    precip_ice_frac = 0.0_r8
+ 
     do k=1,pver
       do i=1,ncol
         cloud_frac(i,k) = min(ast(i,k)+deepcu(i,k),1.0_r8)
```

Comment:
we should probably add a total cloud fraction output which is min(1, max(ice_cloud_frac,liq_cld_frac)+deepcu).

Reply:
I second this!